frontend/src/container/NewDashboard/DashboardSettings/General/index.tsx

```diff
@@ -37,27 +27,35 @@ function GeneralDashboardSettings({
 
 	const { t } = useTranslation('common');
 
-	const onSaveHandler = useCallback(() => {
-		const dashboard = selectedDashboard;
-		// @TODO need to update this function to take title,description,tags only
-		updateDashboardTitleDescriptionTags({
-			dashboard: {
-				...dashboard,
+	const { notifications } = useNotifications();
+
+	const onSaveHandler = (): void => {
+		if (!selectedDashboard) return;
+
+		updateDashboardMutation.mutateAsync(
+			{
+				...selectedDashboard,
 				data: {
-					...dashboard.data,
+					...selectedDashboard.data,
 					description: updatedDescription,
 					tags: updatedTags,
 					title: updatedTitle,
 				},
 			},
-		});
-	}, [
-		updatedTitle,
-		updatedTags,
-		updatedDescription,
-		selectedDashboard,
-		updateDashboardTitleDescriptionTags,
-	]);
+			{
+				onSuccess: (updatedDashboard) => {
+					if (updatedDashboard.payload) {
+						setSelectedDashboard(updatedDashboard.payload);
+					}
+				},
+				onError: () => {
+					notifications.error({
+						message: SOMETHING_WENT_WRONG,
```

Comment:
We should be showing actionable error message , something went wrong doesn't give context and no steps to follow

Reply:
yes agreed server is not sending the actionable error message